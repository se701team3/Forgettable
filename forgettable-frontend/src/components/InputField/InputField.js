import React from 'react';
import {useState} from 'react';
import styles from './InputField.module.css';

/**
 * React input field component used for text entry with the possibility of
 * various stylings that can be applied.
 *
 * Note the "inputType" parameter that must be passed in. Pass in 'primary' for
 * primary input style, 'secondary' for the secondary input style, 'multiLine'
 * for the multiline style. See UI design for more details.
 *
 * Function Arguments:
 * "inputLabel" refers to the label associated with the text input component.
 * "inputType" refers to the "inputType" parameter as mentioned above.
 * "placeholder" refers to the placeholder text in the component.
 * "dataType" refers to the data format allowed as input.
 * "inputID" refers to the ID for the data associated with the input.
 * "inputStateValue" refers to the pre-filled value when the input is rendered.
 * "autoFocusState" refers to the autoFocus attribute, parameter is a Boolean.
 * "requiredState" refers to the required attribute, parameter is a Boolean.
 */

export default function InputField({inputLabel, inputType, placeholder,
  dataType, inputID, inputStateValue, autoFocusState, requiredState}) {
  /**
     * Retrieve state of input component to be rendered in a responsive manner.
     */
  const [inputState, setInput] = useState(inputStateValue);

  /**
     * Types allowed for innput component styling.
     */
  const PRIMARY = 'primary';
  const SECONDARY = 'secondary';
  const MULTILINE = 'multiLine';

  return (
    <div>

      {inputType === PRIMARY &&
        <label className={styles.inputLabelPrimary}>{inputLabel}
          <input className={styles.inputFieldPrimary}
            type={dataType}
            name={inputID}
            placeholder={placeholder}
            value={inputState}
            onChange={(event) => setInput(event.target.value)}
            autoFocus={autoFocusState}
            required={requiredState}
          />
        </label>
      }

      {inputType === SECONDARY &&
        <label className={styles.inputLabelSecondary}>{inputLabel}
          <input className={styles.inputFieldSecondary}
            type={dataType}
            name={inputID}
            placeholder={placeholder}
            value={inputState}
            onChange={(event) => setInput(event.target.value)}
            autoFocus={autoFocusState}
            required={requiredState}
          />
        </label>
      }

      {inputType === MULTILINE &&
      <label className={styles.inputLabelMultiLine}>{inputLabel}
        <textarea className={styles.inputFieldMultiLine}
          type={dataType}
          name={inputID}
          placeholder={placeholder}
          value={inputState}
          onChange={(event) => setInput(event.target.value)}
          autoFocus={autoFocusState}
          required={requiredState}
        />
      </label>
      }

    </div>
  );
}
