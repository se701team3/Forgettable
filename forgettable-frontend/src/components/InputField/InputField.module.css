.inputLabelPrimary {
    font-size: var(--text-medium);
    font-weight: var(--font-medium);
    text-align: left;
    display: flex;
    flex-direction: column;
}

.inputLabelSecondary {
    font-size: var(--text-medium);
    font-weight: var(--font-medium);
    text-align: center;
    vertical-align: middle;
}

.inputLabelMultiLine {
    font-size: var(--text-medium);
    font-weight: var(--font-bold);
    text-align: left;
    display: flex;
    flex-direction: column;
}

.inputFieldPrimary{
    font-size: var(--text-medium);
    width: 375px;
    height: 40px;
    border-radius: var(--radius-btn);
    margin-top: 5px;
    padding: 10px;
}

.inputFieldSecondary{
    font-size: var(--text-medium);
    width: 245px;
    height: 40px;
    border-radius: var(--radius-btn);
    margin-left: 10px;
    padding: 10px;
}

.inputFieldMultiLine {
    font-family: 'Poppins', sans-serif;
    font-size: var(--text-small);
    width: 780px;
    height: 350px;
    border-radius: var(--radius-btn);
    margin-top: 5px;
    padding: 20px;
    resize: none;
}