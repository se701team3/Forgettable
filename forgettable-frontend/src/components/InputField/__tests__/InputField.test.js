import React from 'react';
import InputField from '../InputField';
import {render, cleanup} from '@testing-library/react';
import testRenderer from 'react-test-renderer';

afterEach(cleanup);

it('renders primary input field correctly', () => {
  const primaryInput = testRenderer.create(
      <InputField inputLabel={'PRIMARY'}
        inputType={'primary'}
        placeholder={'PRIMARY TEXT'}
        dataType={'text'}
        inputID={'PrimaryInputField'} />).toJSON();
  expect(primaryInput).toMatchSnapshot();
});

it('renders secondary input field correctly', () => {
  const secondaryInput = testRenderer.create(
      <InputField inputLabel={'SECONDARY'}
        inputType={'secondary'}
        placeholder={'SECONDARY TEXT'}
        dataType={'text'}
        inputID={'SecondaryInputField'} />).toJSON();
  expect(secondaryInput).toMatchSnapshot();
});

it('renders multiLine input field correctly', () => {
  const multiLineInput = testRenderer.create(
      <InputField inputLabel={'MULTILINE'}
        inputType={'multiLine'}
        placeholder={'MULTILINE TEXT'}
        dataType={'text'}
        inputID={'MultiLineInputField'} />).toJSON();
  expect(multiLineInput).toMatchSnapshot();
});


