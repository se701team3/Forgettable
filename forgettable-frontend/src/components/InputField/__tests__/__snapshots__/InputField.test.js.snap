// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiLine input field correctly 1`] = `
<div>
  <label
    className="inputLabelMultiLine"
  >
    MULTILINE
    <textarea
      className="inputFieldMultiLine"
      name="MultiLineInputField"
      onChange={[Function]}
      placeholder="MULTILINE TEXT"
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`renders primary input field correctly 1`] = `
<div>
  <label
    className="inputLabelPrimary"
  >
    PRIMARY
    <input
      className="inputFieldPrimary"
      name="PrimaryInputField"
      onChange={[Function]}
      placeholder="PRIMARY TEXT"
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`renders secondary input field correctly 1`] = `
<div>
  <label
    className="inputLabelSecondary"
  >
    SECONDARY
    <input
      className="inputFieldSecondary"
      name="SecondaryInputField"
      onChange={[Function]}
      placeholder="SECONDARY TEXT"
      type="text"
      value=""
    />
  </label>
</div>
`;
